Content.Server/Chemistry/EntitySystems/SolutionAreaEffectSystem.cs

```diff
@@ -30,7 +30,7 @@ private void OnReactionAttempt(EntityUid uid, SolutionAreaEffectComponent compon
                 return;
 
             // Prevent smoke/foam fork bombs (smoke creating more smoke).
-            foreach (var effect in args.Reaction.Effects)
+            foreach (var effect in args.Reaction.StartEffects)
```

Comment:
This needs to be adapted so that reactions that end in smoke/foam don't slip through.
I might need to implement smoke/foam pooling as a derivative of puddles so producing more smoke in the smoke cloud just adds to the existing smoke cloud instead of spawning infinite smoke. This would also require cutting smoke multiplying reagents so....

Reply:
Foam prototype has slippery by default, also foam and smoke SolutionAreaEffect component are some of the oldest code in the codebase so if you run into weird shit that might explain it